core/consensus/grandpa/impl/vote_crypto_provider_impl.hpp

```diff
@@ -16,7 +16,7 @@ namespace kagome::consensus::grandpa {
    public:
     ~VoteCryptoProviderImpl() override = default;
 
-    VoteCryptoProviderImpl(crypto::Ed25519Keypair keypair,
+    VoteCryptoProviderImpl(boost::optional<crypto::Ed25519Keypair> keypair,
```

Comment:
[minor]
```suggestion
    VoteCryptoProviderImpl(const boost::optional<crypto::Ed25519Keypair> keypair&,
```

Reply:
It is inappropriate here, because keypair is injected by nullable sptr, and this constructor can be called with temp boost::optional